gtsam/inference/EliminateableFactorGraph.h

```diff
@@ -217,60 +215,108 @@ namespace gtsam {
 
     /** Compute the marginal of the requested variables and return the result as a Bayes net.  Uses
      *  COLAMD marginalization ordering by default
-     *  @param variables Determines the variables whose marginal to compute, if provided as an
-     *         Ordering they will be ordered in the returned BayesNet as specified, and if provided
-     *         as a KeyVector they will be ordered using constrained COLAMD.
-     *  @param function Optional dense elimination function, if not provided the default will be
-     *         used.
+     *  @param variables Determines the *ordered* variables whose marginal to compute, 
+     *         will be ordered in the returned BayesNet.
+     *  @param function Optional dense elimination function.
      *  @param variableIndex Optional pre-computed VariableIndex for the factor graph, if not
-     *         provided one will be computed. */
+     *         provided one will be computed. 
+     */
     std::shared_ptr<BayesNetType> marginalMultifrontalBayesNet(
-      boost::variant<const Ordering&, const KeyVector&> variables,
+      const Ordering& variables,
+      const Eliminate& function = EliminationTraitsType::DefaultEliminate,
+      OptionalVariableIndex variableIndex = {}) const;
+
+    /** Compute the marginal of the requested variables and return the result as a Bayes net.  Uses
+     *  COLAMD marginalization ordering by default
+     *  @param variables Determines the variables whose marginal to compute,
+     *         will be ordered using constrained COLAMD.
```

Comment:
Might be worth extra-clarifying that this version will NOT use the key vector ordering, since KeyVector kind of implies an ordering (unlike KeySet), e.g.
```c++
//  @param variables Determines _which_ variables whose marginal to compute,
//           will be ordered using constrained COLAMD.  Will NOT use
//           the KeyVector ordering; use `Ordering(variables)` to specify the variable ordering.
```

Or something like that (same with the other overloads)

Reply:
Also good!